components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemOffer.vue

```diff
@@ -106,21 +106,27 @@ const props = defineProps<{
 }>()
 
 const { apiInstance } = useApi()
-const { urlPrefix, tokenId } = usePrefix()
+const { urlPrefix } = usePrefix()
 const { $route, $i18n, $buefy } = useNuxtApp()
 const { transaction, status, isLoading } = useTransaction()
 const { accountId } = useAuth()
 const { decimals } = useChain()
-const identityStore = useIdentityStore()
 const root = ref<Vue<Record<string, string>>>()
 const connected = computed(() => Boolean(accountId.value))
 
-const balance = computed<string>(() => {
-  if (urlPrefix.value == 'rmrk' || urlPrefix.value == 'ksm') {
-    return identityStore.getAuthBalance
-  }
-  return identityStore.getTokenBalanceOf(tokenId.value)
+const balance = ref<string>('0')
+onMounted(() => {
+  fetchBalance()
```

Comment:
`identityStore.fetchBalance`

Reply:
It dose not work actually. It will fetch the balance of bsx token on bsx. : ) 
That's why I delete this part.

Reply:
I don't think your solution will work on `snek` chain `bsx => '1'; snek => '5'`
We need to find a way to proper fetch balance on each chain, maybe rework `auth.balance` from pinia store to have something like: `auth.balance[prefix][token]`

Reply:
I have changed the id `1` to variable `tokenId`, so that it would fetch balance properly on other chains, including snek. Reworking `auth.balance[prefix][token]` on identityStore would bring too many changes, which seems unnecessary in this case.